multi/stm32l4-multi/rtc.c

```diff
@@ -232,6 +241,124 @@ int rtc_setAlarm(rtctimestamp_t *timestamp)
 }
 
 
+static int rtc_getLastStorage(uint32_t *lastID)
+{
+	uint32_t id[2], valid = 0, num;
+	int ret;
+
+	id[0] = *(rtc_common.base + BACKUP1_ID_REG);
+	id[1] = *(rtc_common.base + BACKUP2_ID_REG);
+
+	if (ID_IS_VALID(id[0])) {
+		valid |= 1;
+	}
+
+	if (ID_IS_VALID(id[1])) {
+		valid |= 2;
+	}
+
+	switch (valid) {
+		case 0: /* No valid records */
+			ret = -1;
+			num = 0;
+			break;
+
+		case 1: /* Only record #1 is valid */
+			ret = 0;
+			num = ID_VALUE(id[0]);
+			break;
+
+		case 2: /* Only record #2 is valid */
+			ret = 1;
+			num = ID_VALUE(id[1]);
+			break;
+
+		default: /* Both records are valid */
+			if (ID_VALUE(id[0]) > ID_VALUE(id[1])) {
+				ret = 0;
+				num = ID_VALUE(id[0]);
+			}
+			else {
+				ret = 1;
+				num = ID_VALUE(id[1]);
+			}
+			break;
+	}
+
+	if (lastID != NULL) {
+		*lastID = num;
+	}
+
+	return ret;
+}
+
+
+int rtc_storeBackup(const void *buff, size_t bufflen)
+{
+	uint32_t idno;
+	int index = (rtc_getLastStorage(&idno) == 0) ? 1 : 0;
+	int idreg = (index == 0) ? BACKUP1_ID_REG : BACKUP2_ID_REG;
+	int retval = -EINVAL;
+	size_t i;
+	uint32_t tmp[BACKUP_PAYLOAD_SZ / sizeof(uint32_t)] = { 0 };
+
+	if ((buff != NULL) || (bufflen != 0) || (bufflen <= BACKUP_PAYLOAD_SZ)) {
+		mutexLock(rtc_common.lock);
```

Comment:
I guess it should be:
```suggestion
	if ((buff != NULL) && (bufflen != 0) && (bufflen <= BACKUP_PAYLOAD_SZ)) {
		mutexLock(rtc_common.lock);
```

Reply:
Whoops! Forgot to change that when I was reversing the logic. Thank you